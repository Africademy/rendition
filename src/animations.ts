import { css, keyframes } from 'styled-components';

const fadeIn = keyframes`
	from {
		opacity: 0;
	}

	to {
		opacity: 1;
	}
`;

const fadeOut = keyframes`
	from {
		opacity: 1;
	}

	to {
		opacity: 0;
	}
`;

export const rotate360 = keyframes`
  0% {transform: rotate(0deg)}
  100% {transform: rotate(360deg)}
`;

export const animations = css`
	.fadeIn {
		animation: ${fadeIn};
	}

	.fadeOut {
		animation: ${fadeOut};
	}

	.animated.faster {
		animation-duration: 500ms;
	}
`;
