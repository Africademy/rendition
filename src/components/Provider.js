import * as React from 'react'
import styled, { ThemeProvider } from 'styled-components'
import defaultTheme from '../theme'

const Base = styled.div`
  font-family: ${props => props.theme.font};
`

const Provider = ({ theme, ...props }) => {
  return (
    <ThemeProvider theme={theme || defaultTheme}>
      <Base {...props} />
    </ThemeProvider>
  )
}

export default Provider
