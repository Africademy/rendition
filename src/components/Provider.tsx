import * as React from 'react';
import styled, { StyledFunction, ThemeProvider } from 'styled-components';
import { DefaultProps, Theme } from '../common-types';
import defaultTheme from '../theme';

export interface ThemedBaseProps extends DefaultProps {
	theme?: Theme;
}

const Base = (styled.div as StyledFunction<DefaultProps>)`
  font-family: ${props => props.theme.font};
`;

const Provider = ({ theme, ...props }: ThemedBaseProps) => {
	return (
		<ThemeProvider theme={theme || defaultTheme}>
			<Base {...props} />
		</ThemeProvider>
	);
};

export default Provider;
