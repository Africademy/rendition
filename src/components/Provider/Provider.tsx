import {
	Grommet
} from 'grommet'
import * as React from 'react';

const Provider = ({ theme, ...props }: any) => {
	return (
		<Grommet theme={theme} {...props} />
	);
};

export default Provider;
