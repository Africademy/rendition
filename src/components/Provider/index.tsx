import { Grommet } from 'grommet';
import merge from 'lodash/merge';
import * as React from 'react';
import styled from 'styled-components';
import { DefaultProps, Theme } from '../../common-types';
import defaultTheme from '../../theme';
import { px } from '../../utils';
import { BreakpointProvider } from './BreakpointProvider';

const Base = styled(Grommet)`
	font-family: ${props => props.theme.font};
	font-size: ${props => px(props.theme.fontSizes[1])};

	h1,
	h2,
	h3,
	h4,
	h5,
	h6,
	button {
		font-family: ${props => props.theme.titleFont};
	}
`;

const Provider = ({ theme, ...props }: ThemedProvider) => {
	const providerTheme = merge(defaultTheme, theme);
	return (
		<BreakpointProvider breakpoints={providerTheme.breakpoints}>
			<Base theme={providerTheme} {...props} />
		</BreakpointProvider>
	);
};

export interface ThemedProvider extends DefaultProps {
	theme?: Partial<Theme>;
}

export default Provider;
