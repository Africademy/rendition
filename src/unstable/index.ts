import { Async } from './components/Async';
import { default as Form } from './components/Form/index';

console.warn(
	'You are using unstable Rendition components! These components are experimental and are not recommended for use in a production environment.',
);

export { Async, Form };
