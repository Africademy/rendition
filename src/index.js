import { default as Banner } from './components/Banner'
import { default as Button } from './components/Button'
import { default as CodeWithCopy } from './components/CodeWithCopy'
import { default as DeleteButton } from './components/DeleteButton'
import { default as DeviceStatusGauge } from './components/DeviceStatusGauge'
import { default as Divider } from './components/Divider'
import { default as Filters } from './components/Filters'
import { default as Fixed } from './components/Fixed'
import { default as Input } from './components/Input'
import { default as Modal } from './components/Modal'
import { default as PineTypes } from './components/PineTypes'
import { default as ProgressBar } from './components/ProgressBar'
import { default as Provider } from './components/Provider'
import { default as SchemaSieve } from './components/Filters/SchemaSieve'
import { default as Select } from './components/Select'
import { default as Text } from './components/Text'
import { default as Textarea } from './components/Textarea'
import { default as Image } from './components/Image'
import { default as Heading } from './components/Heading'
import { default as Container } from './components/Container'
import { default as DropDownButton } from './components/DropDownButton'
import { default as Link } from './components/Link'

import { Flex, Box } from './components/Grid'

export {
  Banner,
  Box,
  Button,
  CodeWithCopy,
  Container,
  DeleteButton,
  DeviceStatusGauge,
  Divider,
  DropDownButton,
  Filters,
  Fixed,
  Flex,
  Heading,
  Link,
  Input,
  Image,
  Modal,
  PineTypes,
  Provider,
  ProgressBar,
  SchemaSieve,
  Select,
  Text,
  Textarea
}
