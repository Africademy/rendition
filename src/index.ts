import { default as Button } from './components/Button';
import { default as Provider } from './components/Provider';

export {
  Button,
  Provider,
};
