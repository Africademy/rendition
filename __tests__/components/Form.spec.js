/* globals expect, describe, it */
import Promise from 'bluebird'
import { mount } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'
import sinon from 'sinon'
import Provider from '../../src/components/Provider'
import { Form } from '../../src/unstable/'

const schema = {
  type: 'object',
  properties: {
    Name: {
      title: 'Pokemon Name',
      type: 'string'
    }
  }
}

describe('Form component', () => {
  it('should match the stored snapshot', () => {
    const component = renderer.create(
      <Provider>
        <Form schema={schema} />
      </Provider>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  describe('hideSubmitButton property', () => {
    it('should render a submit button if the property is not specified', () => {
      const component = mount(
        <Provider>
          <Form schema={schema} />
        </Provider>
      )

      expect(component.find('button')).toHaveLength(1)

      component.unmount()
    })

    it('should render a submit button if the property is falsey', () => {
      const component = mount(
        <Provider>
          <Form schema={schema} hideSubmitButton={false} />
        </Provider>
      )

      expect(component.find('button')).toHaveLength(1)

      component.unmount()
    })

    it('should not render a submit button if the property is truthey', () => {
      const component = mount(
        <Provider>
          <Form schema={schema} hideSubmitButton />
        </Provider>
      )

      expect(component.find('button')).toHaveLength(0)

      component.unmount()
    })
  })

  describe('submitButtonText property', () => {
    it('should change the text value of the submit button', () => {
      const submitText = 'Click me to submit this form'
      const component = mount(
        <Provider>
          <Form schema={schema} submitButtonText={submitText} />
        </Provider>
      )

      expect(component.find('button').text()).toEqual(submitText)

      component.unmount()
    })

    it('should accept a JSX element as a value', () => {
      const submitText = 'Click me to submit this form'
      const submitElement = <span>{submitText}</span>

      const component = mount(
        <Provider>
          <Form schema={schema} submitButtonText={submitElement} />
        </Provider>
      )

      expect(component.find('button').text()).toEqual(submitText)

      component.unmount()
    })
  })

  describe('submitButtonProps property', () => {
    it('should set submit button props', () => {
      const submitButtonProps = {
        className: 'custom-button-class'
      }

      const component = mount(
        <Provider>
          <Form schema={schema} submitButtonProps={submitButtonProps} />
        </Provider>
      )

      expect(component.find('button').hasClass(submitButtonProps.className)).toEqual(true)

      component.unmount()
    })
  })

  describe('onFormSubmit property', () => {
    it('should be called when the submit button is clicked', () => {
      const value = 'Squirtle'
      const callback = sinon.spy()
      const component = mount(
        <Provider>
          <Form schema={schema} onFormSubmit={callback} />
        </Provider>
      )

      const input = component.find('input')
      input.simulate('change', { target: { value } })
      component.update()

      return Promise.delay(150).then(() => {
        component.find('form').simulate('submit')
        expect(callback.callCount).toEqual(1)
        expect(callback.getCall(0).args[0].formData).toEqual({ Name: value })
      })
    })
  })

  describe('onFormChange property', () => {
    it('should be called when an input field is changed', () => {
      const value = 'Squirtle'
      const callback = sinon.spy()
      const component = mount(
        <Provider>
          <Form schema={schema} onFormChange={callback} />
        </Provider>
      )

      const input = component.find('input')
      input.simulate('change', { target: { value } })
      component.update()

      return Promise.delay(150).then(() => {
        expect(callback.callCount).toEqual(1)
        expect(callback.getCall(0).args[0].formData).toEqual({ Name: value })
      })
    })
  })

  describe('value property', () => {
    it('should set the value of the relevant input field', () => {
      const value = { Name: 'Squirtle' }
      const component = mount(
        <Provider>
          <Form schema={schema} value={value} />
        </Provider>
      )

      const input = component.find('input')
      expect(input.props().value).toEqual('Squirtle')
    })

    it('should set the value if the value is changed after the component mounts', () => {
      const value = { Name: 'Squirtle' }
      const component = mount(
        <Form schema={schema} />
      )

      component.setProps({ value })

      const input = component.find('input')
      expect(input.props().value).toEqual('Squirtle')
    })
  })
})
