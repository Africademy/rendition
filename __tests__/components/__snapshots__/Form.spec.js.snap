// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component should match the stored snapshot 1`] = `
.c9 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 1px solid #00AEEF;
  border-radius: 20px;
  color: #444444;
  padding: 4px 30px;
  font-size: 14px;
  line-height: 1.5;
  background: #00AEEF;
  color: #444444;
  border-radius: 20px;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
}

.c9:hover {
  box-shadow: 0px 0px 0px 2px #00AEEF;
}

.c1 {
  font-family: 'Nunito',Arial,sans-serif;
  font-size: 14px;
  line-height: 1.5;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c8 {
  font-family: CircularStd,Arial,sans-serif;
  font-weight: 400;
  height: 36px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: white;
}

.c8 svg {
  color: inherit !important;
  font-size: 0.875em;
}

.c8:disabled {
  cursor: not-allowed;
}

.c8:hover:enabled,
.c8:focus:enabled,
.c8:active:enabled {
  box-shadow: none;
  background: hsl(196.29999999999995,100%,37.5%);
  border-color: hsl(196.29999999999995,100%,37.5%);
  color: white;
  opacity: undefined;
}

.c3 {
  box-sizing: border-box;
}

.c4 {
  margin-bottom: 8px;
}

.c7 {
  border-radius: 3px;
  height: 36px;
  font-size: inherit;
  border: 1px solid #c6c8c9;
  padding: 0px 16px;
  box-sizing: border-box;
}

.c7[disabled] {
  background-color: #f4f4f4;
  cursor: not-allowed;
}

.c7:focus:enabled {
  border-color: #2A506F;
}

.c7:hover:enabled {
  box-shadow: 0 0 4px 1px rgba(0,0,0,0.1);
}

.c7::-webkit-input-placeholder {
  color: #c6c8c9;
}

.c7::-moz-placeholder {
  color: #c6c8c9;
}

.c7:-ms-input-placeholder {
  color: #c6c8c9;
}

.c7::placeholder {
  color: #c6c8c9;
}

.c7[type='checkbox'] {
  height: auto;
  font-size: 14px;
}

.c7[type='date' i]::-webkit-inner-spin-button,
.c7[type='datetime-local' i]::-webkit-inner-spin-button,
.c7[type='month' i]::-webkit-inner-spin-button,
.c7[type='time' i]::-webkit-inner-spin-button,
.c7[type='week' i]::-webkit-inner-spin-button {
  height: 20px;
  margin-top: 7px;
}

.c6 {
  width: 100%;
}

.c0 {
  font-family: 'Nunito',Arial,sans-serif;
}

.c0 h1,
.c0 h2,
.c0 h3,
.c0 h4,
.c0 h5,
.c0 h6,
.c0 label,
.c0 button {
  font-family: CircularStd,Arial,sans-serif;
}

.c5 {
  font-size: 11px;
  color: #252629;
  text-transform: uppercase;
  display: inline-block;
  margin-bottom: 6px;
}

.c2 fieldset {
  margin: 0;
  padding: 0;
  border: 0;
}

.c2 .panel-danger,
.c2 .error-detail {
  font-size: 14px;
  margin-top: 4px;
  margin-bottom: 4px;
  color: #FF423D;
}

<div
  className="c0 c1"
>
  <div
    className="c2 c3"
  >
    <form
      className="rjsf"
      noValidate={false}
      onSubmit={[Function]}
    >
      <div
        className="form-group field field-object rendition-form__field--root c4 c3"
      >
        <fieldset>
          <div
            className="form-group field field-string rendition-form__field--root_Name c4 c3"
          >
            <label
              className="control-label c5"
              htmlFor="root_Name"
            >
              Pokemon Name
            </label>
            <input
              autoFocus={false}
              className="c6 c7"
              disabled={false}
              id="root_Name"
              label="Pokemon Name"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder=""
              readOnly={false}
              required={false}
              type="text"
              value="Bulbasaur"
            />
            <div />
          </div>
        </fieldset>
        <div />
      </div>
      <button
        className="c8 c9"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
